pkg/tenant/db/members.go

```diff
@@ -52,11 +53,73 @@ func (m *Member) UnmarshalValue(data []byte) error {
 	return msgpack.Unmarshal(data, m)
 }
 
-// CreateMember adds a new Member to the database.
+func (m *Member) Validate() error {
+	memberName := m.Name
+
+	if strings.ContainsAny(string(memberName[0]), "0123456789") {
+		return ErrNumberFirstCharacter
+	}
+
+	if strings.ContainsAny(memberName, " !\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~") {
+		return ErrSpecialCharacters
+	}
```

Comment:
Since the methods have been added to each model, I decided to keep doing that for now. Also, the models could change. 

The link didn't work for me and I wasn't able to find exactly what was referenced. Could you try resending?

Reply:
Can you create a follow up story to focus on refactoring the name checks in a way that reduces the repeated code between the different files? I can add some details on the regex there.

I do think it's a good idea to have different validation methods for the different models (to reduce coupling between them), but we might be able to standardize the input checking a bit.